File: Core/Entities/RepairAggregate/RepairRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using Core.Entities.RepairAggregate;

namespace Core.Entities
{
    public class RepairRequest : BaseEntity
    {
       
        public string CustomerEmail { get; set; }
        public string ProductType { get; set; }

        public string ProductBrand { get; set; }
        public string ProblemInfo { get; set; }
        public DateTimeOffset RequestDate { get; set; } = DateTimeOffset.Now;
        public string Address { get; set; }
        public string City { get; set; }
        public string PostalCode { get; set; }
        public string FullName { get; set; }
        public string PhoneNumber { get; set; }

        public string DaysAvailability { get; set; }
        public string PaymentMethods { get; set; }
        public RepairRequestStatus Status { get; set; } = RepairRequestStatus.Unseen;


        public ICollection<TaskRequest> TaskRequest { get; set; }
        public ICollection<TaskRepairman> TaskRepairman { get; set; }
    }
}

File: client/src/app/shared/models/repairRequest.ts
export interface IRepairRequest{
    id: number;
    productInfo: string;
    address: string;
    status: string;
    requestDate: string;
    daysAvailability: IDaysAvailable[];
    paymentMethods: IPaymentMethods[];
    repairmanTask: string;
    subtoTal: number;
    total: number;
    acceptedServiceCall: boolean;
    reasonToReject: string;
    customerEmail: string;
    city: string;
    postalCode: string;
    fullName: string;
    phoneNumber: string;
    productType: string;
    productBrand: string;
    problemInfo: string;
    workPerformed: string;
    repairmanId: number;
    price: number;
    isEmergency: boolean;
    
}

export interface IDaysAvailable {
    id: number;
    name:string;
}

export interface IPaymentMethods {
    id: number;
    name:string;
}
